lxd/network/driver_bridge.go

```diff
@@ -777,16 +777,14 @@ func (n *bridge) setup(oldConfig map[string]string) error {
 
 	// Configure IPv4 firewall (includes fan).
 	if n.config["bridge.mode"] == "fan" || !shared.StringInSlice(n.config["ipv4.address"], []string{"", "none"}) {
-		if n.DHCPv4Subnet() != nil && n.hasIPv4Firewall() {
+		if n.hasDHCPv4() && n.hasIPv4Firewall() {
 			// Setup basic iptables overrides for DHCP/DNS.
 			err = n.state.Firewall.NetworkSetupDHCPDNSAccess(n.name, 4)
 			if err != nil {
 				return err
 			}
-		}
 
-		// Attempt a workaround for broken DHCP clients.
-		if n.hasIPv4Firewall() {
+			// Attempt a workaround for broken DHCP clients.
```

Comment:
@stgraber any reason you can think of why we would need to add the legacy DHCP client checksum fix firewall rule if DHCPv4 isn't enabled on the fan (as it did do this previously)?

Reply:
Nope, that rule isn't needed if we're not running a DHCP server.